### PR TITLE
Remove release links from the readme

### PR DESCRIPTION
**Pull request by lbernick**
# Changes
This information is easy to find on Github and creates more work
for the person doing the release. It also frequently becomes out of date.

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- n/a Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- n/a Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- n/a Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,114 +37,16 @@ Tekton Pipelines are **Typed**:
...
-
-_See [our API compatibility policy](api_compatibility_policy.md) for info on the
-stability level of the API._
-
-_See [our Deprecations table](docs/deprecations.md) for features that have been
-deprecated and the earliest date they'll be removed._
+### Releases
+
+- See [our releases page](https://github.com/tektoncd/pipeline/releases) to find documentation
+for each release.
```
Comment (afrittoli) [inline README.md:43]: This is slightly easier for the maintainers but an extra click or two for our users... not sure we should remove it?
For the out-of-date issue, we could automate it?

Comment (lbernick) [reply README.md:43]: I like the idea of automating it but don't think I have the bandwidth for that right now. I am not too concerned about the extra click or two but happy to close this PR if you'd like to keep the table around.

Comment (abayer) [reply README.md:43]: I'd prefer to keep it around until we can automate it, yeah. Seems like something we could do in `hack/update-codegen.sh` maybe?

Comment (lbernick) [reply README.md:43]: Sounds good -- created #5247

**Comment by lbernick** [comment]
closing, as we want to keep this around but automate it

**Pull request closed without merging**